src/api/app/models/concerns/staging_project.rb

```diff
@@ -142,35 +142,26 @@ def cache_problems
   end
 
   def state
-    return :empty unless staged_requests.exists?
-    return :unacceptable if unacceptable_state?
-    return :review if review_state?
-    general_state
-  end
-
-  def unacceptable_state?
-    untracked_requests.present? || staged_requests.obsolete.present?
-  end
-
-  def review_state?
-    general_state == :acceptable && missing_reviews.present?
+    return :empty if staged_requests.blank?
+    return :unacceptable if untracked_requests.present? || staged_requests.obsolete.present?
```

Comment:
[nitpick] `staged_requests.obsolete` is active record and `present?` load it while `exists?` not

Reply:
Good catch!